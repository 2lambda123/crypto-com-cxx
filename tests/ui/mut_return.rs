#[cxx::bridge]
mod ffi {
    unsafe extern "C++" {
        type Thing;

        fn f(t: &Thing) -> Pin<&mut CxxString>;
        unsafe fn g(t: &Thing) -> Pin<&mut CxxString>;
    }
}

fn main() {}
