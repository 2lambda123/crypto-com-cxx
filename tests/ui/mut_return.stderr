error: &mut return type is not allowed unless there is a &mut argument
 --> $DIR/mut_return.rs:6:9
  |
6 |         fn f(t: &Thing) -> Pin<&mut CxxString>;
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: &mut return type is not allowed unless there is a &mut argument
 --> $DIR/mut_return.rs:7:16
  |
7 |         unsafe fn g(t: &Thing) -> Pin<&mut CxxString>;
  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
